pkg/sql/physicalplan/physical_plan.go

```diff
@@ -294,15 +294,18 @@ func (p *PhysicalPlan) AddNoGroupingStageWithCoreFunc(
 
 // MergeResultStreams connects a set of resultRouters to a synchronizer. The
 // synchronizer is configured with the provided ordering.
+// forceSerialization determines whether the streams are forced to be serialized
+// (i.e. whether we don't want any parallelism).
 func (p *PhysicalPlan) MergeResultStreams(
 	resultRouters []ProcessorIdx,
 	sourceRouterSlot int,
 	ordering execinfrapb.Ordering,
 	destProcessor ProcessorIdx,
 	destInput int,
+	forceSerialization bool,
 ) {
 	proc := &p.Processors[destProcessor]
-	if len(ordering.Columns) == 0 || len(resultRouters) == 1 {
+	if (len(ordering.Columns) == 0 && !forceSerialization) || len(resultRouters) == 1 {
```

Comment:
nit: I find this conditional pretty hard to read. could you make this a 3 way conditional maybe, or find another way to simplify it?